custom_logging/middleware.py

```diff
@@ -0,0 +1,64 @@
+"""
+Copyright 2019 ShipChain, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import threading
+
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+from django.utils.deprecation import MiddlewareMixin
+
+CURRENT_THREAD = threading.current_thread()
+
+
+class UserOrganizationMiddleware(MiddlewareMixin):
+
+    def process_request(self, request):
+        # We put this import here to avoid import error on start up
+        from apps.authentication import PASSIVE_JWT_AUTHENTICATION, passive_credentials_auth
+
+        user_id, org_id = None, None
+        req_auth = request.headers._store.get('authorization')  # pylint: disable=protected-access
+        username = request.POST.get('username')
+
+        if req_auth:
```

Comment:
This feels like reinventing the wheel here - at this point in the request lifecycle I am pretty sure the JWT token should already be decoded by the other middlewares. Can you just use the apps.permissions.get_user() function to get the user/org ID?

Reply:
That's also what I thought but at that point of the process `request.user` is just `AnonymousUser`. That's why I had to go through these strange operations.

Reply:
Is that for every request or just unauthenticated endpoints? According to the docs, since this middleware is at the bottom, the Authentication middleware should have been run first (since it is before it in the middleware ordering):
https://docs.djangoproject.com/en/2.2/topics/http/middleware/#middleware-order-and-layering

So maybe there is another issue..